docs/import/missing-and-suppressed-entities.md

```diff
@@ -28,3 +28,9 @@ The skipped entities are being logged with a `WARN` type and a specitic error me
 ```
 2020-03-06 11:11:25.031  WARN 29784 --- [           main] u.a.e.a.m.importer.ObjectsImporter       : Accession not found in ENA ERS3377874
 ```
+
+# Handling exceptions
+
+During the earlier stages of development, import of Study and Analysis entities was defined to be `@Transactional`, meaning that if any error occurs during import, all changes to the database would be rolled back. This would guarantee a clean but possibly incomplete state after import, where studies and analyses would either be imported completely, or not imported at all.
+
+However, test runs demonstrated that doing a transactional import decreases performance by at least a factor of 10, which is critical. For EVA use case, since the goal is to import all data for all studies, even a single unhandled exception during import means that the import as a whole was a failure and hence the results are unusable in production. Hence, there is no reason to do a transactional import anyway.
```

Comment:
So if a specific import fails, how is the data rolled back in production?

Reply:
You keep asking good questions! Yes, if the import fails, there's currently no way to roll it back. One possible solution here would be to use two databases and to run the import on one while another one serves as production, and to alternate between them. It may be an ugly hack, but for now we don't really have any other way to proceed—if we do it in transactions, the import will just take too much time to be reasonably used. I have amended the documentation about this in https://github.com/EBIvariation/metadata-ws/pull/86/commits/3f3755f4b2643a1f0190e53179842da217cbf607.

Reply:
I agree that this is not optimal; and maybe in the future we can spend more time on this and either improve performance of the import pipeline, or design some other solution to this problem.